### PR TITLE
Data preprocessing: data loader and encode/decoder

### PR DESCRIPTION
**Pull request by JackRegueiro**
Example usage with the `Tiny` class:

```python
from torch.utils.data import DataLoader

if __name__ == "__main__":
    # Parameters
    block_size = 128
    batch_size = 32
    num_epochs = 5

    # Load dataset and tokenizer
    dataset, tokenizer = get_tiny_shakespeare_data(block_size)
    vocab_size = tokenizer.vocab_size

    # DataLoader for batching
    dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=True)

    # Initialize model
    model = Tiny(vocabulary_size)

    # ... train
```

### CODE REVIEW

File: src/data.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        - y (torch.Tensor): A tensor of size `block_size` containing the target sequence,
+        which is offset by 1 token relative to `x`.
+
+        Explanation:
+        - `block_size`: The length of the sequence used as input to the model.
+        - `y` is shifted by one position relative to `x` because the model is trained
+        to predict the next token in the sequence for each position in `x`.
+        """
+        x = self.data[idx:idx+self.block_size]
+        y = self.data[idx+1:idx+self.block_size+1]
```
Comment (jannymongkol) [inline src/data.py:36]: Will we need any padding toward the end of the dataset, in case y's length is 1 less than x's?

**Review by jannymongkol — APPROVED** [review]
